commit 82fba24ad88ae22ee6314b54f419597c14814cf3 (refs/changes/59/4259/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-06-28T12:58:13-07:00 (1 year, 3 months ago)
    
    TRITON-519 cns service crash when vm does not have an owner_uuid
    Reviewed by: Alex Wilson <alex.wilson@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 057fea1..78b1241 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -5,11 +5,15 @@
 -->
 
 <!--
-    Copyright (c) 2015, Joyent, Inc.
+    Copyright (c) 2018, Joyent, Inc.
 -->
 
 # triton-cns
 
-## 1.0.0 (not yet released)
+## 0.1.1
+
+- TRITON-519 fix cns service crash when vm does not have an owner_uuid
+
+## 0.1.0
 
 - Lots of things
diff --git a/lib/ufds-filter.js b/lib/ufds-filter.js
index 6a55fba..51b8f66 100644
--- a/lib/ufds-filter.js
+++ b/lib/ufds-filter.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 module.exports = UfdsFilter;
@@ -51,9 +51,19 @@ util.inherits(UfdsFilter, stream.Transform);
 
 UfdsFilter.prototype._transform = function (vm, enc, cb) {
 	assert.object(vm, 'vm');
-	assert.string(vm.owner_uuid, 'vm.owner_uuid');
+	assert.optionalString(vm.owner_uuid, 'vm.owner_uuid');
 
 	var self = this;
+
+	if (!vm.owner_uuid) {
+		self.log.warn({
+			vm: vm.uuid
+		}, 'no owner_uuid for vm, dropping');
+		self.emit('drop', vm);
+		cb();
+		return;
+	}
+
 	this.getUser(vm.owner_uuid, function (err, owner) {
 		if (err) {
 			self.log.warn({
diff --git a/package.json b/package.json
index 3f7ef51..5137946 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "triton-cns",
-  "version": "0.1.0",
+  "version": "0.1.1",
   "description": "Trition container naming service",
   "main": "index.js",
   "scripts": {
